mlir/lib/IR/Block.cpp

```diff
@@ -349,6 +352,23 @@ SuccessorRange::SuccessorRange(Operation *term) : SuccessorRange() {
     base = term->getBlockOperands().data();
 }
 
+bool Block::isReachable(Block *other, ArrayRef<Block *> except) {
+  assert(getParent() == other->getParent() && "expected same region");
+  SmallVector<Block *> worklist(succ_begin(), succ_end());
+  SmallPtrSet<Block *, 16> visited;
```

Comment:
What about having the "except" parameters be a `SmallPtrSetImpl<Block *>` passed by copy (user should move) and use it to store the visited elements.
That way the user can still pre-populate it with blocks to skip, but we have one less check and one less data-structure around.